var/spack/repos/builtin/packages/py-globus-cli/package.py

```diff
@@ -26,10 +26,3 @@ class PyGlobusCli(PythonPackage):
     depends_on("py-click@8", type=("build", "run"))
     depends_on("py-jmespath@1.0.1", type=("build", "run"))
     depends_on("py-packaging@17:", type=("build", "run"))
-    # According to the developers, these requirements are implicit
-    # for py-globus-sdk, but they are listed explicitly "in case
-    # the underlying lib ever changes"
-    depends_on("py-requests@2.19.1:2", type=("build", "run"))
-    depends_on("py-pyjwt@2.0.0:2+crypto", type=("build", "run"))
-    depends_on("py-cryptography@3.3.1:3.3", type=("build", "run"))
-    depends_on("py-typing-extensions@4:", type=("build", "run"), when="^python@:3.10")
```

Comment:
Just to confirm, this one definitely should not be here?  I ask because it seems to be separately referenced in a `setup.py` comment:

```
        # depend on the latest version of typing-extensions on python versions which do
        # not have all of the typing features we use
        'typing_extensions>=4.0;python_version<"3.11"',
```

Reply:
I'll punt to @climbfuji for original thinking on the inclusion.

For my removal, the logic was "I tried to install _with_ all the extras in place, and got a failure (which I attributed to possible differences between Python versions on @climbfuji's HPC clusters and my pet Ubuntu) - so let's see if it works _without_ any extras".  So I commented them all out and everything worked.

This may very well have been excessive, and perhaps we do need to keep this particular line if it isn't really an extra. I am most definitely open for comments and suggestions from more experienced folks.

Reply:
As a side note, the prerequisite (`py-globus-sdk`) does have `py-typing-extensions` listed, so we are probably in the clear?
```console
    depends_on("python@3.6:", type=("build", "run"))
    depends_on("py-setuptools", type="build")
    depends_on("py-requests@2.19.1:2", type=("build", "run"))
    depends_on("py-pyjwt@2.0.0:2+crypto", type=("build", "run"))
    depends_on("py-cryptography@3.3.1:3.3,3.4.1:", when="@3.7:", type=("build", "run"))
    depends_on("py-cryptography@2:3.3,3.4.1:3.6", when="@:3.0", type=("build", "run"))
    depends_on("py-typing-extensions@4:", when="@3.25:", type=("build", "run"))
```

Reply:
Originally I didn't have the typing extensions line, it came about as part of the code review for merging the package into spack develop. I don't think we need it of `py-globus-sdk` already takes care of that.